template/src/services/wallets/walletconnect/walletConnectClient.tsx

```diff
@@ -52,77 +52,52 @@ export const openWalletConnectModal = async () => {
 };
 
 class WalletConnectWallet implements WalletInterface {
-  private accountId() {
-    // Need to convert from walletconnect's AccountId to hashgraph/sdk's AccountId because walletconnect's AccountId and hashgraph/sdk's AccountId are not the same!
-    return AccountId.fromString(dappConnector.signers[0].getAccountId().toString());
-  }
-
-  // can be replaced when walletconnect provides a signer that satisfies Transaction.executeWithSigner
-  private async signAndExecuteTransaction(transaction: Transaction) {
-    const params: SignAndExecuteTransactionParams = {
-      signerAccountId: `::${this.accountId().toString()}`, // dApps seem to expect two colons in front of the signerAccountId, I'm not sure why. Hoping this gets cleaned up by wallets and walletconnect.
-      transactionList: transactionToBase64String(transaction)
-    };
-    /**
-     * this is not working as expected according to walletconnect's type definitions for dappConnector.signAndExecuteTransaction
-     * 
-     * For HashPack, needed to put in try-catch because hashpack was throwing execptions in dappConnector.signAndExecuteTransaction
-     * For Blade, dappConnector.signAndExecuteTransaction does not throw, but result.result is always undefined. So everywhere that this result is used, I had to add something like `txResult ? txResult.transactionId : null`
-     * Basically for either wallet, the transactionId is not usable.
-     */
-    try {
-      const result = await dappConnector.signAndExecuteTransaction(params);
-      return result.result;
-    } catch {
-      return null;
-    }
-  }
+  private get signer() {
+    return dappConnector.signers[0];
+  } 
 
-  // can be replaced when walletconnect provides a signer that satisfies Transaction.freezeWithSigner
-  private freezeTx(transaction: Transaction) {
-    const nodeAccountIds = hederaClient._network.getNodeAccountIdsForExecute();
-    return transaction
-      .setTransactionId(TransactionId.generate(this.accountId()))
-      .setNodeAccountIds(nodeAccountIds)
-      .freeze();
+  private get accountId() {
+    // Need to convert from walletconnect's AccountId to hashgraph/sdk's AccountId because walletconnect's AccountId and hashgraph/sdk's AccountId are not the same!
+    return AccountId.fromString(this.signer.getAccountId().toString());
   }
 
   async transferHBAR(toAddress: AccountId, amount: number) {
+    console.log(this.accountId);
     const transferHBARTransaction = new TransferTransaction()
-      .addHbarTransfer(this.accountId(), -amount)
+      .addHbarTransfer(this.accountId, -amount)
       .addHbarTransfer(toAddress, amount);
 
-    const frozenTx = this.freezeTx(transferHBARTransaction);
-    const txResult = await this.signAndExecuteTransaction(frozenTx);
+    await transferHBARTransaction.freezeWithSigner(this.signer);
```

Comment:
since the implementation of `this.signer` is now just `return dappConnector.signers[0]`, I think for completeness you should add a n explicit check to see if it is present, before it s used. e.g. `if (!this.signer) { /* handle error gracefully */ }`